tests/conformance_dsl/fragment.rs

```diff
@@ -408,18 +408,51 @@ impl FragmentImpl for TopLevel {
         let mut buffer = Vec::with_capacity(1024);
         let mut writer = Writer::new(config, buffer)?;
 
-        for elem in self.elems.as_slice() {
-            writer.write(ProxyElement(elem, ctx))?;
-        }
+        self.write(ctx, &mut writer)?;
+
         buffer = writer.close()?;
         Ok(buffer)
     }
 
-    fn write<E: ion_rs::Encoding, O: std::io::Write>(
+    fn write<E: Encoding, O: std::io::Write>(
```

Comment:
🪧 The comment below explains the problem I encountered and the workaround that's now in place.